packages/frontend/src/hooks/pivotTable/pivotQueryResults.ts

```diff
@@ -130,10 +130,10 @@ export const pivotQueryResults = ({
             const metric = metrics[nMetric];
 
             const indexRowValues: PivotValue[] = indexDimensions
-                .map<PivotValue>((d) => ({
+                .map<PivotValue>((fieldId) => ({
                     type: 'value',
-                    fieldId: metric.fieldId,
-                    value: row[d].value,
+                    fieldId: fieldId,
```

Comment:
I was referring to a wrong fieldId. instead of dimension id I was referring to metric id.
same change below